### PR TITLE
fix(hasura-storage-js): fix upload response status code check

### LINKED ISSUE #2189: Storage upload in browser does not handle response status code correctly
**Describe the bug**
Storage upload in browser does not handle response status code correctly on this line:
https://github.com/nhost/nhost/blob/73230eb35a7f435142ec659c06a75aef8b214fcf/packages/hasura-storage-js/src/utils/upload.ts#L90

**To Reproduce**
Upload invalid file from the browser - the error response will be returned in the `fileMetadata` field, as if the upload succeeded.

**Expected behavior**
Upstream error from the storage service is returned in the `error` field.

### PR DESCRIPTION
**Pull request by dddenis**
Fixes #2189

### CODE REVIEW

**Comment by onehassan** [comment]
@dddenis Can you please check Allow edits from maintainers so I can add the changeset!

**Comment by dddenis** [comment]
Hey @onehassan , it is already checked.

**Comment by onehassan** [comment]
@dddenis It seems that I still get permission denied somehow

image

**Comment by dddenis** [comment]
Strange, it's been on from the beginning and I've just toggled it off and on again:
Screenshot 2023-08-21 at 20 44 29

I've also added changeset in case you can not commit, let me know if it's ok or if I need to change anything.

File: packages/hasura-storage-js/src/utils/upload.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export const fetchUpload = async (
     xhr.responseType = 'json'
 
     xhr.onload = () => {
-      if (xhr.status < 200 && xhr.status >= 300) {
+      if (xhr.status < 200 || xhr.status >= 300) {
         return resolve({
           fileMetadata: null,
           error: { error: xhr.statusText, message: xhr.statusText, status: xhr.status }
```
Comment (onehassan) [inline packages/hasura-storage-js/src/utils/upload.ts:93]: The error here is in the response

```suggestion
          error: xhr.response?.error ?? xhr.response,
          message: xhr.response?.error?.message ?? xhr.response,
          status: xhr.status
```

Comment (dddenis) [reply packages/hasura-storage-js/src/utils/upload.ts]: Updated https://github.com/nhost/nhost/pull/2190/commits/08fe4cd65f717ace5ff0f6b435a4b33bf37eaf77

**Review by dbarrosop — APPROVED** [review]

**Pull request merged**